run.go

```diff
@@ -86,16 +120,142 @@ func createGrpcServer() *grpc.Server {
 	return service.GRPCServer
 }
 
-func startPrometheusServer() {
+func startPrometheusServer(started func()) {
 	service.PrometheusServer = &http.Server{Addr: service.PrometheusConfig.Address()}
 
 	http.Handle("/metrics", promhttp.Handler())
 	logrus.Infof("Prometheus metrics at http://%s/metrics", service.PrometheusConfig.Address())
 
+	service.prometheusRunning = true
+	started()
+
 	go func() {
 		if err := service.PrometheusServer.ListenAndServe(); err != nil {
 			// cannot panic, because this probably is an intentional close
 			logrus.Errorf("Prometheus http server: ListenAndServe() error: %s", err)
 		}
 	}()
 }
+
+func startProbeServer() {
```

Comment:
Could the probe server code live in a separate package?

Reply:
Yeah good shout. There's a few things in here which could do with being moved out the root

Reply:
done